File: binarysearch/gen.py
import random
import json

num = 500000
x = 0
data = []

while True:
    if x > num:
        break
    temp = random.randint(0, num)
    if temp not in data:
        data.append(temp)
        x += 1

with open("data1.json", 'w') as f:
    json.dump(data, f)

File: binarysearch/gen.js
var fs = require("fs");
const arr = [];
while (arr.length < 100000) {
  const r = Math.floor(Math.random() * 100000) + 1;
  if (arr.indexOf(r) === -1) {
    arr.push(r);
  }
}
console.log(arr);

fs.writeFile("data1.json", JSON.stringify(arr), function err() {
  console.log("ayo");
});
